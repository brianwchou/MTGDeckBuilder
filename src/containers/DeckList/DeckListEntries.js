import React from 'react'
import DeckListEntry from 'containers/DeckList/DeckListEntry';

export default function DeckListEntries({data, type, counts, getCardInfo}) {
    let entries = data.map((entry, index)=> {
        return <DeckListEntry index={index} card={entry} getCardInfo={getCardInfo} count={counts[entry.name]} />
    }) 
    return (entries.length) ? (
        <>
            <div className="type_padding">{type}</div>
            <div>
                {entries}
            </div>
        </>
    ) : null;
}
