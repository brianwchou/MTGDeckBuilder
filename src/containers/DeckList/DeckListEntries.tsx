import React from 'react'
import DeckListEntry from './DeckListEntry'
import { CardInfo } from '../../common/types';

interface CardCount {
    [index: string]: number
}

type DeckListEntriesProps = {
    data: Array<CardInfo>,
    type: string,
    counts: CardCount,
    getCardInfo(card: CardInfo, name: string): void,
}

export default function DeckListEntries({data, type, counts, getCardInfo}: DeckListEntriesProps) {
    let entries = data.map((entry: CardInfo, index: number)=> {
        return <DeckListEntry key={index} card={entry} getCardInfo={getCardInfo} count={counts[entry.name]} />
    }) 
    return (entries.length) ? (
        <>
            <div className="type_padding">{type}</div>
            <div>
                {entries}
            </div>
        </>
    ) : null;
}
