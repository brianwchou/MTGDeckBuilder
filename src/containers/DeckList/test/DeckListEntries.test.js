import React from 'react'
import DeckListEntries from 'containers/DeckList/DeckListEntries';
import renderer from 'react-test-renderer'
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('DeckListEntry', () => {
    
    test('rendering correctly', () => {
        const mockNoCard = []
        const tree = renderer
        .create(<DeckListEntries data={mockNoCard}/>)
        .toJSON();

    expect(tree).toMatchSnapshot();
    });

    test('rendering correctly with cards', () => {
        const mockCards = [{name:"thing in the ice"}]
        const mockType = "creature"
        const mockCounts = {thing_in_the_ice: 2}
        const tree = renderer
        .create(<DeckListEntries data={mockCards} counts={mockCounts} type={mockType} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
    });
});