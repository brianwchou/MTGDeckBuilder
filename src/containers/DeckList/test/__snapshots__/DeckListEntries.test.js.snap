// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckListEntry rendering correctly 1`] = `null`;

exports[`DeckListEntry rendering correctly with cards 1`] = `
Array [
  <div
    className="type_padding"
  >
    creature
  </div>,
  <div>
    <div
      className="flex alignment"
    >
      <div
        className="flex count"
      >
        x
        <button
          name="increment"
          onClick={[Function]}
        >
          +
        </button>
      </div>
      <div
        className="flex name"
      >
        thing in the ice
        <button
          name="decrement"
          onClick={[Function]}
        >
          -
        </button>
        <button
          name="maybe"
          onClick={[Function]}
        >
          maybe
        </button>
      </div>
    </div>
  </div>,
]
`;
